virtcontainers/network.go

```diff
@@ -593,6 +639,45 @@ func initNetworkCommon(config NetworkConfig) (string, bool, error) {
 		return path, true, nil
 	}
 
+	// Check if the network namespace requested is the same as the runtime
+	var evalNS, nsPath, currentNsPath string
+	var err error
+
+	// Here we are trying to resolve the path but it fails because
+	// namespaces links don't really exist. For this reason, the
+	// call to EvalSymlinks will fail when it will try to stat the
+	// resolved path found. As we only care about the path, we can
+	// retrieve it from the PathError structure.
+
+	if strings.HasPrefix(config.NetNSPath, "/proc") && strings.HasSuffix(config.NetNSPath, "/ns/net") {
```

Comment:
This block is effectively self-contained so I think you should put it into a new function so you can say:

```
if hostNetworkingRequested(config.NetNSPath) {
    return "", false, fmt.Errorf("Host networking requested, not supported by kata-runtime")
}
```

That will of course also allow you to fully unit test the new functionality.